src/plots/cartesian/axes.js

```diff
@@ -2306,15 +2319,18 @@ axes.drawOne = function(gd, ax, opts) {
     // this happens when updating matched group with 'missing' axes
     if(!mainPlotinfo) return;
 
+    ax._shift = setShiftVal(ax, axShifts);
+
     var mainAxLayer = mainPlotinfo[axLetter + 'axislayer'];
     var mainLinePosition = ax._mainLinePosition;
+    var mainLinePositionShift = (ax.shift !== false && ax.shift !== undefined) ? mainLinePosition += ax._shift : mainLinePosition;
```

Comment:
Using `+=` operator on this line while also setting another variable is quite confusing.

Reply:
Good point -- I can really simplify this.